### PR TITLE
Fix MSVC compile issues in CloudComposer

### PR DESCRIPTION
**Pull request by SunBlack**
Fix MSVC compile issues in CloudComposer. This required to build pcl_cc_tool_interface as static library, because current PCL_EXPORTS macro doesn't work with classes using Q_OBJECT (see #2708).

cloud_composer is now using pcl_cc_tool_interface to prevent some cpps need to be compiled for both.

### CODE REVIEW

File: 2d/CMakeLists.txt [hunk 1]
```diff
@@ -33,7 +33,7 @@ if(build)
         set(VTK_IO_TARGET_LINK_LIBRARIES vtkCommon vtkWidgets vtkIO vtkImaging)
     endif()
 
-    include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include" ${VTK_INCLUDE_DIRECTORIES})
+    include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include")
```
Comment (taketwo) [inline 2d/CMakeLists.txt]: Why is this?

File: apps/cloud_composer/CMakeLists.txt [hunk 2]
```diff
@@ -45,157 +45,169 @@ PCL_SUBSUBSYS_DEPEND(build "${SUBSYS_NAME}" "${SUBSUBSYS_NAME}" DEPS ${SUBSUBSYS
...
-     #Supervoxels
-    set(VSP_DEPS pcl_octree pcl_segmentation)
-    set(VSP_SOURCES tools/supervoxels.cpp "include/pcl/${SUBSYS_NAME}/${SUBSUBSYS_NAME}/tools/impl/supervoxels.hpp")
-    set(VSP_HEADERS "include/pcl/${SUBSYS_NAME}/${SUBSUBSYS_NAME}/tools/supervoxels.h")
-    define_composer_tool (supervoxels "${VSP_SOURCES}" "${VSP_HEADERS}" "${VSP_DEPS}")
-
-    # Add to the compound apps target
-    list(APPEND PCL_APPS_ALL_TARGETS ${EXE_NAME})
-    set(PCL_APPS_ALL_TARGETS ${PCL_APPS_ALL_TARGETS} PARENT_SCOPE)
+if(NOT build)
```
Comment (taketwo) [inline apps/cloud_composer/CMakeLists.txt]: Hard to see, does this modify the lines, or only changes indentation? If latter, please put in a separate commit.

File: apps/cloud_composer/ComposerTool.cmake [hunk 3]
```diff
@@ -19,11 +19,10 @@ function(define_composer_tool TOOL_NAME TOOL_SOURCES TOOL_HEADERS DEPS)
   add_definitions(-DQT_NO_DEBUG)
   add_definitions(-DQT_SHARED)
 
-  add_dependencies(${TOOL_TARGET} pcl_cc_tool_interface ${DEPS})
```
Comment (taketwo) [inline apps/cloud_composer/ComposerTool.cmake]: Why is this removed?

File: io/CMakeLists.txt [hunk 4]
```diff
@@ -329,7 +329,7 @@ if(build)
 
     add_definitions(${VTK_DEFINES})
     PCL_ADD_LIBRARY("${LIB_NAME}" "${SUBSYS_NAME}" ${srcs} ${incs} ${compression_incs} ${impl_incs} ${OPENNI_INCLUDES} ${OPENNI2_INCLUDES})
-    target_include_directories(${LIB_NAME} PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include" ${VTK_INCLUDE_DIRECTORIES})
+    target_include_directories(${LIB_NAME} PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
```
Comment (taketwo) [inline io/CMakeLists.txt]: Why is this removed?

File: 2d/CMakeLists.txt (hunk 1 above)
Comment (SunBlack) [reply 2d/CMakeLists.txt]: I don't find related website now again, but as far as I read VTK has since version 6 imported targets, so it is not necessary anymore. And because first compile issues were VTK related I started to play a bit around with it. It is a leftover, but because I don't know if I spend more time on VTK in CMake I didn't removed it before commit.

File: apps/cloud_composer/CMakeLists.txt (hunk 2 above)
Comment (SunBlack) [reply apps/cloud_composer/CMakeLists.txt]: This just reduce indentation of following lines. Seems Github does not support viewing diff Side-by-Side (compared to GitLab), so it is indeed a bit hard to see changes here.

File: apps/cloud_composer/ComposerTool.cmake (hunk 3 above)
Comment (SunBlack) [reply apps/cloud_composer/ComposerTool.cmake]: There are no more targets than in `target_link_libraries`, so it is already done automatically by CMake. A explicit call to `add_dependencies` is only necessary if you have dependencies which CMake can't see.

File: io/CMakeLists.txt (hunk 4 above)
Comment (SunBlack) [reply io/CMakeLists.txt]: See above (disucssion in 2d/CMakeLists.txt).

**Comment by SunBlack** [comment]
> If the Windows build is now fixed, can we enable CloudComposer on Azure?

I don't know: See your own reason why this is not build on Azure (https://github.com/PointCloudLibrary/pcl/issues/2618#issuecomment-441791923)

File: 2d/CMakeLists.txt (hunk 1 above)
Comment (taketwo) [reply 2d/CMakeLists.txt]: I see. Can we remove this (and similar change in `io`) for now? If somebody (or you) have time in future to confirm that this is not necessary anymore, it would be nice to merge a single PR that fixes all PCL modules, not just one or two.

**Comment by taketwo** [comment]
Ah right, thanks for the heads-up. We can enable this when (if) we switch to container-based Windows builds.

File: 2d/CMakeLists.txt (hunk 1 above)
Comment (SunBlack) [reply 2d/CMakeLists.txt]: There are no more occurrences of `VTK_INCLUDE_DIRECTORIES` in whole PCL project. But i can remove it from this PR if you want.

Comment (taketwo) [reply 2d/CMakeLists.txt]: OK let's keep it here, but would be great if you could put this in a separate commit with an explanatory message.

Comment (SunBlack) [reply 2d/CMakeLists.txt]: I currently wait for build results to see, if it really works with VTK 6.2, because I'm using 8.2 ;).

File: apps/cloud_composer/CMakeLists.txt (hunk 2 above)
Comment (SunBlack) [reply apps/cloud_composer/CMakeLists.txt]: General question to this: Is this change in general ok (in a separate PR)?
```
if(build)
// Whole remaining CMake code
endif() //endif at end of file
```

```
if(build)
  return()
endif()
// Whole remaining CMake code
```

Comment (taketwo) [reply apps/cloud_composer/CMakeLists.txt]: Yes, I'm fine with this, less nesting -> easier to read.

File: 2d/CMakeLists.txt (hunk 1 above)
Comment (SunBlack) [reply 2d/CMakeLists.txt]: Moved separate commit into separate PR #2713 ;)
